docs/coding-guidelines/scripting-guide.md

```diff
@@ -0,0 +1,14 @@
+Scripting Guide
+===============
+
+Generally, for script files (.sh, .ps1 & .cmd), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
+
+PowerShell
+-----------------
+
+### -NoProfile
+Unlike Unix or Linux based shells, when PowerShell is invoked by default it executes [profile scripts](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles), even on Unix or Linux systems. This feature may cause side effects which may affect executing code. To guard against this, it is a requirement to always invoke PowerShell with the `-NoProfile` option. For example, the following PowerShell command in valid:
```

Comment:
This is just one of the many best practices for writing and using PowerShell scripts. I do not think we want to maintain list of best practices for writing shell scripts in this repo, the shell scripts are just a small fraction of it.

Can we just link to a good list of Powershell best practices from the other doc?

Reply:
Have updated per your suggestions. Have linked to the [Scripting devblog](https://devblogs.microsoft.com/scripting) which, although slightly hard to find, there is an article about using [-NoProfile](https://devblogs.microsoft.com/scripting/powertip-launch-powershell-without-a-profile). This was the best resources I could find regarding "official" PowerShell best practices.